system/Commands/Utilities/Optimize.php

```diff
@@ -52,15 +52,31 @@ final class Optimize extends BaseCommand
      *
      * @var string
      */
-    protected $usage = 'optimize';
+    protected $usage = 'optimize [-c] [-l] [-d]';
+
+    /**
+     * The Command's options
+     *
+     * @var array<string, string>
+     */
+    protected $options = [
+        'c' => 'Enable config caching.',
+        'l' => 'Enable locator caching.',
+        'd' => 'Disable config and locator caching.',
+    ];
 
     /**
      * {@inheritDoc}
      */
     public function run(array $params)
     {
+        // Parse options
+        $enableConfigCache  = CLI::getOption('c');
+        $enableLocatorCache = CLI::getOption('l');
+        $disable            = CLI::getOption('d');
+
         try {
-            $this->enableCaching();
+            $this->enableCaching($enableConfigCache, $enableLocatorCache, $disable);
             $this->clearCache();
             $this->removeDevPackages();
```

Comment:
This code is not testable. Because the command always removes dev packages.

Reply:
It seems that, this command previously also did not have tests, while the other commands did.

Reply:
I had mistakenly thought that the command with options only changed the configuration file. 
However, it might be better to modify the configuration file and perform optimization, as is the current behavior.

If so, the test code cannot be written in a simple way, so the way it is written needs to be changed.

Reply:
For example,
```php
<?php

declare(strict_types=1);

/**
 * This file is part of CodeIgniter 4 framework.
 *
 * (c) CodeIgniter Foundation <admin@codeigniter.com>
 *
 * For the full copyright and license information, please view
 * the LICENSE file that was distributed with this source code.
 */

namespace CodeIgniter\Commands\Utilities;

use CodeIgniter\Test\CIUnitTestCase;
use PHPUnit\Framework\Attributes\Group;

/**
 * @internal
 */
#[Group('Others')]
final class OptimizeTest extends CIUnitTestCase
{
    public function testEnableConfigCaching(): void
    {
        $command = new Optimize(service('logger'), service('commands'));

        $enableCaching = $this->getPrivateMethodInvoker($command, 'enableCaching');

        // private function enableCaching(?bool $enableConfigCache, ?bool $enableLocatorCache, ?bool $disable): void
        $enableCaching(true, null, null);

        // Check if config caching is enabled
        $this->assertFileContains('public bool $configCacheEnabled = true;', APPPATH . 'Config/Optimize.php');
    }

    protected function assertFileContains(string $needle, string $filePath): void
    {
        $this->assertFileExists($filePath);
        $this->assertStringContainsString($needle, file_get_contents($filePath));
    }
}
```